cmd/gokr-packer/packer.go

```diff
@@ -1293,6 +1314,34 @@ func logic() error {
 	return nil
 }
 
+func updateWithProgress(prog *progress.Reporter, reader io.Reader, target *updater.Target, logStr string, stream string, optional bool) error {
+	start := time.Now()
+	prog.SetStatus(fmt.Sprintf("update %s", logStr))
+	prog.SetTotal(0)
+
+	if stater, ok := reader.(interface{ Stat() (os.FileInfo, error) }); ok {
+		if st, err := stater.Stat(); err == nil {
+			prog.SetTotal(uint64(st.Size()))
+		}
+	}
+	if err := target.StreamTo(stream, io.TeeReader(reader, &progress.Writer{})); err != nil {
+		if err == updater.ErrUpdateHandlerNotImplemented && optional {
+			log.Printf("target does not support updating %s yet, ignoring", stream)
```

Comment:
Remove the `optional` parameter and return the error here using `return fmt.Errorf("updating %s: %w", logStr, err)`, then use `errors.As()` in the caller to handle the error there.